components/automate-ui/src/app/pages/notification-details/notification-details.component.ts

```diff
@@ -0,0 +1,175 @@
+  NotificationRule,
+  ServiceActionType,
+  RuleType
+} from 'app/entities/notification_rules/notification_rule.model';
+
+type NotificationTabName = 'details';
+
+enum UrlTestState {
+  Inactive,
+  Loading,
+  Success,
+  Failure
+}
+
+@Component({
+  selector: 'app-notification-details',
+  templateUrl: './notification-details.component.html',
+  styleUrls: ['./notification-details.component.scss']
+})
+
+export class NotificationDetailsComponent implements OnInit, OnDestroy {
+  public tabValue: NotificationTabName = 'details';
+  public notification = new NotificationRule('', '', null, '', null, '', false);
+  public updateForm: FormGroup;
+  public saveInProgress = false;
+  public testInProgress = false;
+  public saveSuccessful = false;
+  public hookStatus = UrlTestState.Inactive;
+  public notificationDetailsLoading = true;
+  public targetKeys: string[];
+  public alertTypeKeys: string[];
+  private isDestroyed = new Subject<boolean>();
+
+  constructor(
+    private fb: FormBuilder,
+    private store: Store<NgrxStateAtom>,
+    private layoutFacade: LayoutFacadeService
+  ) {
+
+    this.updateForm = this.fb.group({
+      // Must stay in sync with error checks in notification-details.component.html
+      name: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]],
+      targetType: [[]],
+      ruleType: [[]],
+      // Note that URL here may be FQDN -or- IP!
+      targetUrl: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]]
+    });
+
+    this.store.select(routeParams).pipe(
+      pluck('id'),
+      filter(identity),
+      takeUntil(this.isDestroyed))
+      .subscribe((id: string) => {
+        this.store.dispatch(new GetNotification({ id }));
+      });
+
+    combineLatest([
+      this.store.select(getStatus),
+      this.store.select(notificationFromRoute)
+    ]).pipe(
+      filter(([status, notification]) =>
+      status === EntityStatus.loadingSuccess && !isNil(notification)),
+      takeUntil(this.isDestroyed))
+      .subscribe(([_, notification]) => {
+        this.notification = notification;
+        setTimeout(() => (
+          this.updateForm.controls.name.setValue(this.notification.name),
+          this.updateForm.controls.targetUrl.setValue(this.notification.targetUrl),
+          this.notificationDetailsLoading = false
+        ),
+        200);
+      });
+
+    this.store.pipe(
+      select(updateStatus),
+      filter(state => this.saveInProgress && !pending(state),
+      takeUntil(this.isDestroyed)))
+      .subscribe((state) => {
+        this.saveInProgress = false;
+        this.saveSuccessful = (state === EntityStatus.loadingSuccess);
+        if (this.saveSuccessful) {
+          this.updateForm.markAsPristine();
+        }
+      });
+  }
+
+  public ngOnInit(): void {
+    this.layoutFacade.showSidebar(Sidebar.Settings);
+    this.targetKeys = this.notification.getTargetTypeKeys();
+    this.alertTypeKeys = this.notification.getAlertTypeKeys();
```

Comment:
Thanks for this change (removing function calls from the template)! I notice the exact same fix is  needed in notification_rule.model.ts, though (getAlertTypeKeys() and getTargetTypeKeys()). Could you fix that, too, please?

Reply:
@msorens 
which function needs to fix in notification_rule.model.ts.

Reply:
Exactly the same calls as you did here: not doing function calls getAlertTypeKeys() and getTargetTypeKeys() in the HTML file.

Reply:
@msorens 
you are talking about the notification-form.component.html, we are going to remove this after create a notification modal, so I have not changed this file for now.

Reply:
OK.